### DISCUSSION TITLE
shamefully-hoist=true vs node-linker=hoisted

### DISCUSSION THREAD

**Discussion by hubertkuoch** [pnpm/pnpm#5409] [Q&A] [answered]
What is the difference between those two settings? 

Looks like in both cases, all dependencies are hoisted to the root node_modules. 

This is particularly important to clarify for monorepos containing next.js apps which consume buildable libs since next build webpack config does not resolve symlink properly for server bundle.

**Comment by zkochan** [comment, marked as answer]
shamefully-hoist uses symlinks to hoist.

node-linker=hoisted creates a node_modules w/o symlinks. Similar to the one created by npm or Yarn v1
